src/terrain/clients/data_info.clj

```diff
@@ -77,23 +77,42 @@
   [^String user ^String dir]
   (cr/ensure-created user dir))
 
+(defn get-public-data-user
+  "Returns the anonymous user for public data if a user is not provided"
+  [user paths]
+  (let [paths (if (= (type paths) String)
+                [paths]
+                paths)
+        has-private-paths? (some #(not (string/starts-with? % (cfg/fs-community-data))) paths)
+        request-user (if has-private-paths?
+                       user
+                       (or user "anonymous"))]
```

Comment:
I haven't looked too closely into this yet, but would it be reasonable to let `data-info` validate the permissions on the files instead? The advantage of letting `data-info` do the checks is that it would be possible to use Terrain to access endpoints that are anonymously readable, but outside of the community data collection. The disadvantage is that you'd have to add custom error handling to detect permissions errors and return a 401 in cases where the user wasn't authenticated. It also might take a little bit longer to get 403 responses back if response times are a concern.

Reply:
The possible issue here is that a lot of folders are anonymously accessible due to "send to genome browser" (which, IIRC, shares the parent folders too), but folks normally don't expect those to be "browsable". We ran into the same issue with dav-anon and turned off the index pages for it for that reason.

Reply:
Ah okay. It probably is best to restrict it to community data then.

Reply:
That brings up an interesting point though - do those "send to genome browser" folders become searchable for the anonymous user?  If so, then I might need to rethink some things for the search endpoint so only community data results come up.  I did some mild testing and only saw community data results, but I may not be hitting the right search terms in that case.

Reply:
They become accessible to the anonymous user, but it sounds like we don't want people who aren't logged in to be able to discover that these files exist. My guess is that we don't want these files to be searchable by the anonymous user.

Reply:
To close the loop on this one, we decided this should be added to the `search` service so that it can better handle modifying the query to narrow it down to only community data and also handle contradictory clauses and whatnot.  A ticket was created for that work.